compiler/shared_ast/definitions.ml

```diff
@@ -666,14 +673,20 @@ type scope_info = {
   out_struct_fields : StructField.t ScopeVar.Map.t;
 }
 
+(** In practice, this is a DAG: beware of repeated names *)
+type module_tree = M of module_tree ModuleName.Map.t [@@caml.unboxed]
```

Comment:
Why don't you use an `ocamlgraph` type to represent the DAG then?

Reply:
It wouldn't be convenient for lookup by key, which is the main use-case here: we only have to be careful about repetitions during the initial load